pandas/tests/reshape/merge/test_multi.py

```diff
@@ -837,3 +836,29 @@ def test_join_multi_wrong_order(self):
         )
 
         tm.assert_frame_equal(result, expected)
+
+
+def test_merge_datetime_index_empty_df():
+
+    frame = DataFrame(
+        {
+            "date": [pd.Timestamp("1950-01-01"), pd.Timestamp("1950-01-02")],
+            "panel": ["A", "B"],
+            "data": [1.5, 1.5],
+        }
+    ).set_index(["date", "panel"])
+    other = DataFrame(columns=["date", "panel", "state"]).set_index(["date", "panel"])
+
+    expected = DataFrame(
+        {
+            "date": [pd.Timestamp("1950-01-01"), pd.Timestamp("1950-01-02")],
+            "panel": ["A", "B"],
+            "data": [1.5, 1.5],
+            "state": [None, None],
+        }
+    )
+    expected = expected.set_index(["date", "panel"])
+
+    result = frame.merge(other, how="left", on=["date", "panel"])
+
+    tm.assert_frame_equal(result, expected)
```

Comment:
Removes use of `set_index()` to isolate testing to `merge()`, [as requested here](https://github.com/pandas-dev/pandas/pull/36897#discussion_r500545112).

```suggestion
    index_column_names = ["date", "panel"]
    
    frame = DataFrame(
        data={
            "date": [pd.Timestamp("1950-01-01"), pd.Timestamp("1950-01-02")],
            "panel": ["A", "B"],
            "data": [1.5, 1.5],
        },
        index=index_column_names
    )
    other = DataFrame(
        columns=["date", "panel", "state"],
        index=index_column_names
    )
    actual_result = frame.merge(other, how="left", on=index_column_names)

    expected_result = DataFrame(
        data={
            "date": [pd.Timestamp("1950-01-01"), pd.Timestamp("1950-01-02")],
            "panel": ["A", "B"],
            "data": [1.5, 1.5],
            "state": [None, None],
        },
        index=index_column_names
    )

    tm.assert_frame_equal(actual_result, expected_result)
```